code/WorkInProgress/MechanicMadness.dm

```diff
@@ -2007,6 +2008,19 @@
 				componentSay("Removed : [input.signal]")
 		return
 
+	proc/popitem(var/datum/mechanicsMessage/input)
+		if(level == 2 || !input) return
+		if(current_index <= length(signals))
+			LIGHT_UP_HOUSING
+			sendCurrent(input)
+			signals.Remove(signals[current_index])
+			if(current_index > length(signals))
+				current_index = length(signals) ? length(signals) : 1 // Don't let current_index be 0
+			tooltip_rebuild = 1
+			if(announce)
+				componentSay("Removed : [input.signal]")
```

Comment:
```suggestion
		sendCurrent(input)
		remitem(input)
```
All the same checks and effects are triggered in these two functions.
`sendCurrent()` also modifies the signal `input`, so the correct item will be removed when passed to `remitem()`
Also I tested this on local and it worked well.